### ISSUE TITLE
Question of operating the simple classification model

### ISSUE THREAD

**Issue by designe** [nnstreamer/nnstreamer#3741]
Hi!

I'm currently testing a simple image classification model.

https://github.com/nnstreamer/nnstreamer-example/blob/d1d13bf896d9e378947f869a0f254a27042f1ff3/native/example_image_classification_tflite/nnstreamer_example_image_classification_tflite.py#L71-L79

By referring to the above example, I wrote a below pipeline, then it was working well.

`mobilenet_v1_1.0_224_quant.tflite`
- input_tensor : uint8(dtype), 3:224:224:1(dims)
- output_tensor : MobilenetV1/Prediction/Reshape_1(name), uint8(dtype), 1001:1:1:1(dims)
```
t. ! queue leaky=2 max-size-buffers=2 ! videoconvert ! videorate \
! videoscale ! video/x-raw,width=224,height=224,format=RGB,framerate=5/1 \
! tensor_converter ! tensor_filter framework=tensorflow-lite model=" + self.ai_model + " \
! tensor_sink name=tensor_sink 
```

### Here is my question.
I have a simple image classification tf model what I made.
- input_tensor : input(name), float32(dtype), 3:448:448:1(dims)
- output_tensor : TEST/output(name), float32(dtype), 4:1:1:1(dims)

#### python script
- python3.7
- tensorflow1.14.0
```py3
import cv2
import tensorflow as tf
import numpy as np

model_path = "test.pb"
dgraph = tf.Graph()
with dgraph.as_default():
    od_graph_def = tf.GraphDef()
    with tf.gfile.GFile(model_path, 'rb') as fid:
        serialized_graph = fid.read()
        od_graph_def.ParseFromString(serialized_graph)
        tf.import_graph_def(od_graph_def, name='')
        
sess = tf.Session(graph=dgraph)
input_tensor = dgraph.get_tensor_by_name('input:0')
logit_tensor = dgraph.get_tensor_by_name("TEST/output:0")

def process(image):
    return sess.run(logit_tensor, feed_dict={input_tensor: image})

cap = cv2.VideoCapture('test.mp4')
while(True):
    _, frame = cap.read()
    input_frame = cv2.resize(frame, (448, 448), interpolation = cv2.INTER_AREA)
    input_rgb_frame = cv2.cvtColor(input_frame, cv2.COLOR_BGR2RGB)
    input_source = np.expand_dims(input_rgb_frame, axis=0).astype(np.float32)

    logits2= process(input_source)
    result = logits2[0,0,0]
    print(result)
```
#### Script Result
```bash
[0.04415273 0.00838822 0.9033102  0.01109459]
[0.0586564  0.00825193 0.8687464  0.01489588]
[0.04534589 0.01063451 0.8726463  0.01752995]
[0.04586298 0.01033099 0.87111866 0.01774858]
[0.0782729  0.00657561 0.8320046  0.01695431]
[0.0559187  0.00728175 0.8429163  0.0207606 ]
[0.04843298 0.0073402  0.8421235  0.02431537]
[0.03361611 0.01175327 0.85357785 0.02833812]
[0.03292592 0.00565606 0.8352839  0.03711707]
[0.03655645 0.00413026 0.8115764  0.03802608]
[0.03449119 0.00400977 0.7821078  0.05087843]
[0.03789895 0.00365371 0.7090478  0.07378442]
.
.
```

#### Below is my tested pipeline.
```
"t. ! queue leaky=2 max-size-buffers=2 ! videoconvert ! videorate \
! videoscale ! video/x-raw,width=448,height=448,format=RGB,framerate=5/1 \
! tensor_converter ! tensor_transform mode=arithmetic option=typecast:float32,div:255 \
! tensor_filter framework=tensorflow model=" + self.ai_model + " input=3:448:448:1 inputname=input inputtype=float32 output=4:1:1:1 outputname=TEST/output outputtype=float32 \
! tensor_sink name=tensor_sink "
```
#### pipeline result
```
[9.9997133e-01 9.8049641e-06 2.3931265e-05 1.1533499e-05]
[9.9997139e-01 9.8943710e-06 2.3931265e-05 1.1503696e-05]
[9.9997139e-01 9.8943710e-06 2.3931265e-05 1.1503696e-05]
[9.9997139e-01 9.8943710e-06 2.3931265e-05 1.1503696e-05]
[9.9997139e-01 9.8943710e-06 2.3931265e-05 1.1503696e-05]
[9.9997139e-01 9.8943710e-06 2.3931265e-05 1.1503696e-05]
[9.9997139e-01 9.7751617e-06 2.3931265e-05 1.1563301e-05]
[9.9997139e-01 9.7751617e-06 2.3931265e-05 1.1563301e-05]
[9.9997139e-01 9.7751617e-06 2.3931265e-05 1.1563301e-05]
[9.9997139e-01 9.7751617e-06 2.3931265e-05 1.1563301e-05]
.
.
```

Although they're using same input(mp4), but my tested pipeline's result is different from the result operated by `python script`.
so I am wondering if there is any missing part.

I think the difference with above Mobilenet example is that the type `uint8`(mobilenet) or `float32`(my model). 
Is there anything I need to add more?

**Comment by designe** [comment]
I'm sharing my tensor_sink's `new-data`

```py3
    def on_new_data(self, sink, buffer):
        """Callback for tensor sink signal.
        :param sink: tensor sink element
        :param buffer: buffer from element
        :return: None
        """
        if self.running:
            if buffer.n_memory() != 1:
                return False                                        
            #1. TEST/output: Independent class probability for each classes
            # 4:1:1:1
            probability = buffer.peek_memory(0)
            result, probability_box = probability.map(Gst.MapFlags.READ)
            if result:
                if probability_box.size == 16:
                    probs = np.fromstring(probability_box.data, dtype=np.float32)
                    print(probs)
            
            probability.unmap(probability_box)
```

**Comment by gichan-jang** [comment]
@designe , Thanks for posting the issue.
The difference between python script and pipeline is normalization of the input of the tensor filter.
If the Python script result is correct, changing the pipeline to get the same result.
```
... ! tensor_converter ! tensor_transform mode=typecast option=float32 ! ...

```

**Comment by designe** [comment]
@gichan-jang Thanks for your support!
Now it is very working well :)

 I didn't know how to typecast from `uint8` to `float32`
So I bring tensor_convertor from example usage.

Thanks again 👍
